src/v/redpanda/admin/transform.cc

```diff
@@ -225,6 +236,15 @@ admin_server::deploy_transform(std::unique_ptr<ss::http::request> req) {
             env.insert_or_assign(v["key"].GetString(), v["value"].GetString());
         }
     }
+    model::compression compression{model::compression::none};
+    if (doc.HasMember("compression")) {
+        std::string_view cs = doc["compression"].GetString();
+        try {
+            compression = boost::lexical_cast<model::compression>(cs);
+        } catch (const boost::bad_lexical_cast& e) {
+            vlog(adminlog.debug, "Unrecognized compression mode: '{}'", cs);
```

Comment:
nit: Maybe a more severe error level?  Otherwise the user may not know that they provided a bad compression mode.

nit: Should we exit and return an error to the caller since they violated the API?

Reply:
>Should we exit...

JSON validator should provide API-level error signaling, so this is more of a sanity check in case the enum and the schema fall out of sync during development. 

That said, I think throwing `bad_request` is probably better that a debug log in practice.